lib/ui/results.js

```diff
@@ -0,0 +1,338 @@
+    return url;
+  },
+
+  renderLabel: function(result) {
+    /*
+    Creates and returns an <hbox> element containing the "Recommended" label.
+    */
+    let label = createElement(result.doc, 'hbox', 'label');
+    label.textContent = result.getLabelText(result);
+    return label;
+  },
+
+  _get: function(key) {
+    /*
+    Attempt to return a value from the recommendation data at the passed key
+    string. If not set, falsy, or an empty object, return null;
+    */
+    const val = key.split('.').reduce(function(o, x) {
+      return (typeof o == 'undefined' || o === null) ? o : o[x];
+    }, this._data);
+    let isEmptyObject = function(val) {
+      return typeof val == 'object' && !Object.keys(val).length;
+    };
+    return (typeof val == 'undefined' || !val || isEmptyObject(val)) ? null : val;
+  },
+
+  _firstLetterInBaseDomain: function(result) {
+    /*
+    Returns the first letter of the base domain for the passed result.
+    */
+    let url = result._get('result.url');
+    return result.eTLD.getBaseDomain(result.io.newURI(url, null, null))[0];
+  },
+
+  _getLetterboxColors: function(result) {
+    /*
+    Calculates and returns an object literal containing CSS background and
+    foreground colors for the letterbox from the favicon.
+
+    The background is determined by reading the most prominent color from the
+    favicon. If the brightness of that value is greater than 75%, the
+    foreground is black. Otherwise, it is white.
+    */
+    let color = result._get('enhancements.favicon.color');
+    if (color) {
+      const r = color[0];
+      const g = color[1];
+      const b = color[2];
+      const brightness = ((r * 0.2126 + g * 0.7152 + b * 0.0722) / 255);
+      return {
+        'background': `rgb(${r}, ${g}, ${b})`,
+        'foreground': brightness > 0.75 ? 'black' : 'white'
+      };
+    }
+    return null;
+  },
+
+
+  _getContentMaxWidth: function(result) {
+    /*
+    This method is bad.
+
+    It attempts to calculate the available space for the recommendation content
+    (Title, URL) by subtracting a magic number from the width of the popup.
+
+    That magic number estimates the amount of horizontal space that non-content
+    elements are taking up. The calculation for that number:
+
+      9px recommendation left padding
+     16px favicon
+      8px favicon right margin
+     32px image/letterbox
+      8px image/letterbox right margin
+      8px label left margin
+    100px label width (appx)
+     15px recommendation right padding
+    -----
+    197px
+
+    If the available space is less than 200px, we will grow the popup to ensure
+    that there are at least 200px available for it.
+    */
+    const MIN = 200;
+    const MAGIC = 197;
+    const popupWidth = Math.floor(
+      result.doc.getElementById('PopupAutoCompleteRichResult').width);
+    const available = popupWidth - MAGIC;
+    return available >= MIN ? available : MIN;
+  }
+};
+
+
+
+function TLDResult(data, opts) {
```

Comment:
I agree. How would you recommend doing the imports? Is it okay to just import them into this file, or would you prefer that the `resultsClass` constant is in the main module where the majority of our imports happen?